tfs/reader.py

```diff
@@ -125,25 +144,34 @@ def read_tfs(
     metadata: _TfsMetaData = _read_metadata(tfs_file_path)
 
     if metadata.column_names is None:
-        errmsg = f"No column names in file {tfs_file_path.absolute()}. File not read."
-        raise TfsFormatError(errmsg)
+        raise AbsentColumnNameError(tfs_file_path)
     if metadata.column_types is None:
-        errmsg = f"No column types in file {tfs_file_path.absolute()}. File not read."
-        raise TfsFormatError(errmsg)
+        raise AbsentColumnTypeError(tfs_file_path)
+
+    # The pandas engines do NOT support reading complex numbers, so we take note of which columns are
+    # complex-dtyped and will provide a specific converter function for pandas to use
+    dtypes_dict = dict(zip(metadata.column_names, metadata.column_types))
+    converters: dict[str, Callable] = {}
+
+    if np.complex128 in metadata.column_types:
+        LOGGER.debug("Complex columns detected, reading as strings and casting later")
+        for colname, dtype in zip(metadata.column_names, metadata.column_types):
+            if dtype is np.complex128:
+                converters[colname] = _parse_complex  # will a special converter
+                del dtypes_dict[colname]  # remove to avoid ParserWarning saying we provided both
```

Comment:
**FEATURE**

Hopefully this is clear enough, but here goes:

- Keep track of which columns are complex-dtyped (from metadata reading and dtype identifiers - if the TFS file lies to us through the identifier we are fucked and will crash in the lines doing the reading below).
- Put these in a special dictionary (`converters`) and assign to that entry our custom-made function to load complex-dtyped data.
- Remove the entry from the first dictionnary (with all the "normal" columns).

After these lines we have:
- A `dtypes_dict` dict with all non-complex columns (name as key, dtype to use as value).
- A `converters` dict with all complex columns (name as key, function to use to load it as value).

These are then provided to the pandas reader which uses either its own functions for the loading or our implementation for the complex-columns (it does not have one).